java-api-core/src/com/kinvey/java/store/BaseDataStore.java

```diff
@@ -270,6 +270,15 @@ public T save (T object) throws IOException {
     }
 
     /**
+     * Clear the local cache storage
+     */
+    public void clear() {
+        client.getCacheManager().getCache(getCollectionName(), storeItemType, Long.MAX_VALUE).delete(new Query());
+        client.getSyncManager().getCacheManager().getCache("sync", SyncRequest.class, Long.MAX_VALUE).delete(new Query());
+        client.getSyncManager().getCacheManager().getCache("syncitems", SyncItem.class, Long.MAX_VALUE).delete(new Query());
+    }
+
+    /**
```

Comment:
I think we should check some preconditions here, just as in the `purge` method. Also, can we just call `purge` from this method, in order to clear out the sync queue?

Reply:
Makes sense. Preconditions and using purge were added to clear method.